### PR TITLE
Adding inventory dashboard

### PR DESCRIPTION
**Pull request by ronniegeraghty**
Moving scripts for the collection and formatting of data for the SDK Inventory Dashboard to this repo.

### CODE REVIEW

File: eng/scripts/inventory-dashboard/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Inventory Dashboard
+
+[Dashboard link](https://aka.ms/azsdk/sdk-inventory)
+
+A dashboard for the Azure SDK team to track release libraries across program languages and Azure services. The dashboard uses data from the [Azure SDK Release CSVs](https://github.com/Azure/azure-sdk/tree/main/_data/releases/latest).
+
+## Dashboard
+
+The dashboard is a Power BI dashboard and it's save file is located [here](./dashboard/AzureSDKInventoryDashboard.pbix).
```
Comment (weshaggard) [inline eng/scripts/inventory-dashboard/README.md:9]: We generally avoid committing binary files into git repos. Does this need to be in the repo itself or should it just remain in Power BI?

File: eng/scripts/inventory-dashboard/src/addEmptyDataPoints.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  PackageList,
+  TrackSpecificsDefault,
+  Plane,
+} from "./types";
+import { Logger } from './logger';
+import _serviceNameMap from "../data-and-rules/serviceNameMap.json";
+import path from "path";
+import fs from 'fs';
+const log = Logger.getInstance();
+const specsDirPath = path.join(__dirname, '../../../../../../Code/azure-rest-api-specs/specification');
```
Comment (weshaggard) [inline eng/scripts/inventory-dashboard/src/addEmptyDataPoints.ts:12]: This path is going to be broken when you move it to running it in the repo.

Comment (ronniegeraghty) [reply eng/scripts/inventory-dashboard/src/addEmptyDataPoints.ts:12]: Yeah, I thought as much. Was just there for testing the change locally.
I'm about to start working on getting the scripts added to the pipeline. Any pointers there? 
Specifically:
- Where to add the API specs repo to it is also cloned?
- Adding a step to run the typescript files?
- Getting the created CSV file added to the PR being made?

Comment (weshaggard) [reply eng/scripts/inventory-dashboard/src/addEmptyDataPoints.ts:12]: - Add a clone step similar to https://github.com/Azure/azure-sdk/blob/main/eng/pipelines/version-updater.yml#L80 for the specs repo
- Be sure to do all your steps before the create pr step at https://github.com/Azure/azure-sdk/blob/main/eng/pipelines/version-updater.yml#L58
- Make sure you use the azure-sdk clone at $(AzureSDKClonePath) to do all your reading and writing of files you want to be included in the PR. 
- To run a typescript command in CI you can try to mimic an example like https://github.com/Azure/azure-sdk-for-js/blob/main/eng/pipelines/templates/stages/archetype-js-release.yml#L205-L216

If you have any other questions feel free to ping in IM

Comment (ronniegeraghty) [reply eng/scripts/inventory-dashboard/src/addEmptyDataPoints.ts]: Changed the path to what I believe should still work in the pipeline env

File: eng/scripts/inventory-dashboard/README.md (hunk 1 above)
Comment (ronniegeraghty) [reply eng/scripts/inventory-dashboard/README.md:9]: I've seen us do this with a couple other PowerBI dashboard files, but they are all in an internal ADO repo. That is probably the better spot for it. I'll remove it and get it added there.

File: eng/scripts/inventory-dashboard/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  for (let pkgKey in formattedCSVData) {
+    formattedPackageArr.push(formattedCSVData[pkgKey]);
+  }
+  // Write Formatted Object to CSV File
+  json2csv(formattedPackageArr, (err, csv) => {
+    if (err) {
+      log.err(`Error Writing to CSV. Error: ${err}`);
+    } else if (csv) {
+      fs.writeFileSync(
+        path.join(__dirname, "../FormattedPackagesFromCSVs.csv"),
```
Comment (ronniegeraghty) [inline eng/scripts/inventory-dashboard/src/index.ts:36]: @weshaggard Did we agree on a place in the azure-sdk repo to place the resulting CSV file the dashboard will use?

Comment (weshaggard) [reply eng/scripts/inventory-dashboard/src/index.ts:36]: I think we agreed to create a folder under https://github.com/Azure/azure-sdk/tree/main/_data/releases

**Comment by ronniegeraghty** [comment]
@weshaggard with these changes I believe it's ready to test in the pipeline. If not, let me know what should be changed. If so, I believe you said there was a command I could enter in the comments to test run the pipeline?

**Review by weshaggard — APPROVED** [review]
Looks like a good starting point.

**Pull request merged**